### PR TITLE
Handle go-junit-report v2 changes

### PR DESCRIPTION
**Pull request by draftcode**
This absorbs the difference between go-junit-report v1 and v2. Making
the data backward compatible with v1.

### CODE REVIEW

File: launchable/test_runners/go_test.py [hunk 1]
```diff
@@ -32,6 +37,37 @@ def subset(client):
...
+            return
+
+    default_path_builder = client.path_builder
+
+    def path_builder(case: TestCase, suite: TestSuite, report_file: str) -> TestPath:
+        tp = default_path_builder(case, suite, report_file)
+        for tpc in tp:
+            if 'type' in tpc and 'name' in tpc and tpc['type'] == 'class':
+                # go-junit-report v2 reports full package name. go-junit-report
+                # v2 reports only the last component of the package name. In
```
Comment (Konboi) [inline launchable/test_runners/go_test.py:63]: > go-junit-report v2 reports only the last component of the package name

I think v1 is correct

File: launchable/test_runners/go_test.py [hunk 2]
```diff
@@ -32,6 +37,37 @@ def subset(client):
...
+                # order to make this backward compatible, we align this to v1.
+                tpc['name'] = tpc['name'].split('/')[-1]
```
Comment (Konboi) [inline launchable/test_runners/go_test.py:65]: make sense

**Review by Konboi — APPROVED** [review]
LGTM

File: launchable/test_runners/go_test.py (hunk 1 above)
Comment (draftcode) [reply launchable/test_runners/go_test.py]: Done.

**Pull request merged**